src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Command/Entity/ChooseEntityTypeCommand.php

```diff
@@ -26,6 +26,11 @@ class ChooseEntityTypeCommand
      */
     private $type;
 
+    /**
+     * @var array
+     */
+    private $choices;
```

Comment:
Why is this in the command if this could come directly from the ProtocolChoiceFactory?

Reply:
I always find it hard to load dynamic choices on a form. This is my preferred method. Do you suggest a different approach?

Reply:
Inject the ProtocolChoiceFactory (or a repository) in the form with the constructor.

Reply:
Would that work? To my knowledge, Symfony Types do no support dependency injection.

Reply:
Thanks for pointing out IRL that Form types can actually be injected with dependencies when tagged with `form.type`. 

The additional complexion that further prevents moving this logic to the Type, is the fact that we need to know about the active service in order to determine which protocols may be displayed. 
 
This could be fixed by injecting the authz service on the factory. But that would increase coupling and decrease the cohesion..

I'm still opting for this solution.